File: trustly-android-sdk/src/main/java/net/trustly/android/sdk/util/cid/CidStorage.java
package net.trustly.android.sdk.util.cid;

import android.content.Context;
import android.content.SharedPreferences;

public class CidStorage {

    private static final String CID_STORAGE = "CID_STORAGE";
    public static final String SESSION_CID = "SESSION_CID";
    public static final String CID = "CID";

    public CidStorage() {
        throw new IllegalStateException("Utility class cannot be instantiated");
    }

    public static void saveData(Context context, String preferenceId, String preferenceValue) {
        getSharedPreferences(context).edit().putString(preferenceId, preferenceValue).apply();
    }

    public static String readDataFrom(Context context, String preferenceId) {
        return getSharedPreferences(context).getString(preferenceId, null);
    }

    private static SharedPreferences getSharedPreferences(Context context) {
        return context.getSharedPreferences(CID_STORAGE, Context.MODE_PRIVATE);
    }

}

File: trustly-android-sdk/src/main/java/net/trustly/android/sdk/util/cid/CidStorage.kt
package net.trustly.android.sdk.util.cid

import android.content.Context
import net.trustly.android.sdk.data.TrustlyStorage

object CidStorage {

    private const val CID_STORAGE: String = "CID_STORAGE"
    const val SESSION_CID: String = "SESSION_CID"
    const val CID: String = "CID"

    fun saveData(context: Context, preferenceId: String, preferenceValue: String?) {
        getTrustlyStorage(context).saveData(preferenceId, preferenceValue)
    }

    fun readDataFrom(context: Context, preferenceId: String) =
        getTrustlyStorage(context).readStringDataFrom(preferenceId)

    private fun getTrustlyStorage(context: Context) = TrustlyStorage(context, CID_STORAGE)

}